### ISSUE TITLE
'CompileC' error after React Native Upgrade

### ISSUE THREAD

**Issue by efeakyuz** [facebook/react-native#35082]
### Description

`The following build commands failed:`
`CompileC /Users/username/Library/Developer/Xcode/DerivedData/Aposto-gmqtdupzstzukkcwwycvjhdnwfeh/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/React-Core.build/Objects-normal/x86_64/RCTAppSetupUtils.o /Users/username/file/Dev/mobile/node_modules/react-native/React/AppSetup/RCTAppSetupUtils.mm normal x86_64 objective-c++ com.apple.compilers.llvm.clang.1_0.compiler (in target 'React-Core' from project 'Pods')
(1 failure)
`

I wanted to take a big step from 0.63.3 to 0.70.3. I made all the changes with the upgrade helper. I have done the installations according to x86_64 before, but I wanted to use arm when switching to the new react-native version, maybe there was a problem here, I don't know. I have tried to clean the cache and everything I can do and find on the internet, but nothing of these works.

### Version

0.70.3

### Output of `npx react-native info`

```
System:
    OS: macOS 12.4
    CPU: (8) arm64 Apple M1
    Memory: 1.33 GB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.13.0 - /usr/local/bin/node
    Yarn: 1.22.17 - /usr/local/bin/yarn
    npm: 8.1.0 - /usr/local/bin/npm
    Watchman: 2022.10.24.00 - /opt/homebrew/bin/watchman
  Managers:
    CocoaPods: 1.11.3 - /opt/homebrew/opt/ruby@2.7/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.4, iOS 15.5, macOS 12.3, tvOS 15.4, watchOS 8.5
    Android SDK:
      API Levels: 28, 29, 30, 31, 33
      Build Tools: 29.0.2, 30.0.2, 30.0.3, 31.0.0
      System Images: android-31 | Google APIs ARM 64 v8a
      Android NDK: Not Found
  IDEs:
    Android Studio: 2021.2 AI-212.5712.43.2112.8815526
    Xcode: 13.4/13F17a - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.11 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.1.0 => 18.1.0 
    react-native: ^0.70.3 => 0.70.3 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

### Steps to reproduce

1 - run yarn react-native run-ios
2 - build failed

### Snack, code example, screenshot, or link to a repository

I don't have any code to show, it's not a my code problem

**Comment by Marcoo09** [comment]
Hi @efeakyuz Maybe this thread is useful for you: https://github.com/facebook/react-native/issues/26118#issuecomment-525085809

**Comment by efeakyuz** [comment]
@Marcoo09 I already checked this, nothing came up here :(

**Comment by efeakyuz** [comment]
Problem solved yey. Maybe these reproduce steps will help someone. #26118 (comment) I applied the solution here before, but nothing happened. I tried these steps again. Then I stopped running Xcode with rosetta and came back on arm64. I realized that these kinds of errors are much more difficult to resolve with CLI. so I started getting builds from Xcode and check the errors from there. The first error I saw was the "Flipper is not found". I think I solved it by removing `#use_frameworks!:linkage => :static`, I don't really remember. then I got `FirebaseCore is not found` error. I solved this by adding `pod 'FirebaseCore', :modular_headers => true` to the Podfile. Then `return [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];`
I got error here and I fixed it by converting it to `return [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index"];` Now I can get builds without any problems.

**Comment by EmilianoMorghen** [comment]
@efeakyuz This was actually a good suggestion. 
I was facing the error:
`The following build commands failed: CompileC`
I've tried to load the project into xcode, build and see the errors like you suggested.

I saw that I was missing `AppDelegate.m`. Good because upgrading from react-native@0.67.4 to react-native@0.68.5 change the file from `.m` format to `.mm`.

![image](https://user-images.githubusercontent.com/9059989/213561738-cc9be3e0-190f-4db8-b5d3-97ec1b89d2be.png)
By changing the name of the file from `AppDelegate.m` to `AppDelegate.mm` worked.
